javax/ws/rs/core/MediaTypeTest.cpp

```diff
@@ -0,0 +1,220 @@
+    MediaType mediaType7("application", "*");
+    ASSERT_TRUE(mediaType6.isCompatible(mediaType7));
+    ASSERT_TRUE(mediaType7.isCompatible(mediaType6));
+}
+
+TEST(MediaType, ValueOf) {
+    // MediaType valueOf "application/json", expected type "application", subtype "json"
+    MediaType mediaType = MediaType::valueOf("application/json");
+    ASSERT_STR("application", mediaType.getType().toString());
+    ASSERT_STR("json", mediaType.getSubtype().toString());
+
+    // MediaType valueOf "text/*", expected type "text", subtype "*+plain"
+    MediaType mediaType1 = MediaType::valueOf("text/*+plain");
+    ASSERT_STR("text", mediaType1.getType().toString());
+    ASSERT_STR("*+plain", mediaType1.getSubtype().toString());
+
+    // MediaType valueOf WILDCARD, expected both type and subtype are MEDIA_WILDCARD_TYPE
+    MediaType mediaType3 = MediaType::valueOf(MediaType::WILDCARD);
+    ASSERT_STR(MediaType::MEDIA_TYPE_WILDCARD.toString(), mediaType3.getType().toString());
+    ASSERT_STR(MediaType::MEDIA_TYPE_WILDCARD.toString(), mediaType3.getSubtype().toString());
+
+    // MediType valueOf empty String, expected exception "`type` must not be empty"
+    MediaTypeException exception;
+    try {
+        MediaType mediaType4 = MediaType::valueOf("");
+    } catch (MediaTypeException &e) {
+        exception = e;
+    }
+    ASSERT_STR("`type` must not be empty", exception.getMessage().toString());
+
+    // MediaType valueOf "/", expected exception "/ does not contain subtype after '/'"
+    MediaTypeException exception1;
+    try {
+        MediaType mediaType5 = MediaType::valueOf("/");
+    } catch (MediaTypeException &e) {
+       exception1 = e;
+    }
+    ASSERT_STR("/ does not contain subtype after '/'", exception1.getMessage().toString());
+
+    // MediaType valueOf "text/", expected exception "/ does not contain subtype after '/'"
+    MediaTypeException exception2;
+    try {
+        MediaType mediaType5 = MediaType::valueOf("text/");
+    } catch (MediaTypeException &e) {
+        exception2 = e;
+    }
+    ASSERT_STR("text/ does not contain subtype after '/'", exception2.getMessage().toString());
+
+    // MediaType valueOf "*/html", expected exception wrong type
+    MediaTypeException exception3;
+    try {
+        MediaType mediaType5 = MediaType::valueOf("*/html");
+    } catch (MediaTypeException &e) {
+        exception3 = e;
+    }
+    ASSERT_STR("*/html wildcard type is legal only in '*/*' (all types)", exception3.getMessage().toString());
+}
+
+TEST(MediaType, ToString) {
+
+    // Default constructor, expected WILDCARD
+    MediaType mediaType;
+    mediaType.toString();
+    ASSERT_STR(MediaType::WILDCARD.toString(), mediaType.toString().toString());
+
+    // Constructor with type "application", subtype "*+xml", expected "application/*+xml"
+    MediaType mediaType2("application", "*+xml");
+    ASSERT_TRUE(mediaType2.toString().equals((String)"application/*+xml"));
```

Comment:
Not wrong here
But when it failed, it will not show expected and actual result to compare